### PR TITLE
Bumpallowedpygeofilter

### PR DESCRIPTION
**Pull request by bitner**
Bump allowed version of pygeofilter to include 0.2.*.

### CODE REVIEW

**Review by vincentsarago — APPROVED** [review]
🥳

File: stac_fastapi/pgstac/setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "asyncpg",
     "buildpg",
     "brotli_asgi",
-    "pygeofilter>=0.1,<0.2",
+    "pygeofilter>=0.1,<0.3",
     "pypgstac==0.6.*",
     "pygeoif==0.7",
```
Comment (vincentsarago) [inline stac_fastapi/pgstac/setup.py:21]: In theory we could do 
```
"pygeofilter>=0.2,<0.3"
```
and remove `pygeoif` requirement

**Comment by jtherrmann** [comment]
+1 for getting this merged. We had to pin `pygeofilter==0.2.0` in our repo due to a dependency bug (details at https://github.com/ASFHyP3/asf-stac/issues/6).

**Comment by gadomski** [comment]
Just checking before merge, are there known incompatibilities with `pygeofilter == 0.3`?

**Comment by gadomski** [comment]
Added a manual cherry-pick of a7f5c7568e04736382060de0df4e34539278b094 to ceil fastapi.

**Review by geospatial-jeff — APPROVED** [review]

**Pull request merged**